common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/sftp/SftpSessionFactoryProperties.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.sftp;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.stream.module.file.remote.AbstractRemoteServerProperties;
+
+/**
+ * @author Gary Russell
+ */
+@ConfigurationProperties
+public class SftpSessionFactoryProperties extends AbstractRemoteServerProperties {
+
+	/**
+	 * Resource location of user's private key.
+	 */
+	private String privateKey = "";
+
+	/**
+	 * Passphrase for user's private key.
+	 */
+	private String passPhrase = "";
+
+	/**
+	 * True to allow an unknown or changed key.
+	 */
+	private boolean allowUnknownKeys = false;
+
+	/**
+	 * A SpEL expression resolving to the location of the known hosts file.
+	 */
+	private String knownHostsExpression = null;
```

Comment:
Doesn't matter. What I'm saying is that it would surface a parsing exception as a binder exception (hence appearing alongside other user PEBCAK errors), instead of a context initialization error.
Doesn't matter much if it does not though